scripts/build_binary.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# Normalize to working directory being build root (up one level from ./scripts)
+ROOT=$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )
+cd "${ROOT}"
+
+go run ./internal/pkg/version/gen/generate-version.go
+
+GIT_SHORT_HASH=`git rev-parse --short=7 HEAD`
+TAGGED=`git tag --points-at ${GIT_SHORT_HASH}`
+
+if [ -z "$TAGGED" ]; then
+  GIT_DIRTY=`echo '*'`
+fi
+
+GIT_HASH="$GIT_DIRTY$GIT_SHORT_HASH"
```

Comment:
At a glance this looks somewhat similar to what `git describe` outputs, maybe we can simply use that? The example output from the docs would be something like:
`v1.0.4-14-g2414721`

It's the latest tag `-` number of changes since that tag `-` latest commit hash.

Reply:
oooh, i didn't know about `git describe --tags`! #funFact thanks!